packages/sanes-chrome-extension/src/logic/persona/personamanager.ts

```diff
@@ -0,0 +1,44 @@
+import { Persona } from './persona';
+
+export class PersonaManager {
+  private static instance: Persona | undefined;
+
+  /**
+   * Creates and stores a Persona instance
+   */
+  public static async create(fixedMnemonic?: string): Promise<Persona> {
+    const newInstance = await Persona.create(fixedMnemonic);
+    PersonaManager.instance = newInstance;
+    return newInstance;
+  }
+
+  /**
+   * Gets the global Persona instance
+   *
+   * Throws if instance does not exist, which indicates a bug in the app life cycle.
+   */
+  public static get(): Persona {
+    if (!PersonaManager.instance) {
+      throw new Error('Persona instance does not exist');
```

Comment:
When I follow this pattern I am used to creating the Persona if it does not already exist. But I understand it has downsides:
1. If you recreate it (in the future maybe we can do that) we would need to manage the auto destroy of the previous instance.
2. Harder to trail errors
3. Seeing the only entry point will be a "react" PersonaProvider, is not that bad use there the  annoying try/catch

So if you want to keep as it is, despite is not the "recommended" way, SGTM.

Reply:
Looking into the future, when Persona is either randomly generated, loaded from DB or restored from a mnemonic, I think we will run into all kind of trouble if we don't create the Persona explicitly. Thus I prefer the current solution.